app/api/src/db/sql/utils.py

```diff
@@ -51,7 +56,59 @@ def sorted_path_by_dependency(path_list):
     return new_path_list
 
 
-if __name__ == "__main__":
-    path_list = Path(str(Path().resolve()) + "/src/db/sql/functions").rglob("*.sql")
-    new_path_list = sorted_path_by_dependency(path_list)
-    print(new_path_list)
+def list_functions():
+    query = """
+    SELECT
+    routine_name
+    FROM 
+        information_schema.routines
+    WHERE 
+        routine_type = 'FUNCTION'
+    AND
+        routine_schema = :functions_schema;
+    """
+    query = text(query)
+    with legacy_engine.connect() as session:
+        functions = session.execute(
+            query, {"functions_schema": settings.POSTGRES_FUNCTIONS_SCHEMA}
+        )
+        return [f["routine_name"] for f in functions]
+
+
+def report():
+    function_paths = Path(str(Path().resolve()) + "/src/db/sql/functions").rglob("*.sql")
+    triger_paths = Path(str(Path().resolve()) + "/src/db/sql/triggers").glob("*.sql")
+    files = list(function_paths) + list(triger_paths)
+
+    functions = list_functions()
+    classified_functions = defaultdict(list)
+    not_in_db = []
+    for fn in files:
+        if not fn:
+            continue
+
+        file_name = fn.parts[-1]
+        directory_name = fn.parts[-2]
+        if get_name_from_path(fn) in functions:
+            functions.remove(get_name_from_path(fn))
+        else:
+            not_in_db.append((directory_name, file_name))
+        classified_functions[directory_name].append(file_name)
+
+    for key in classified_functions.keys():
+        print("##", key + "/")
```

Comment:
I have once written some very basic functions for printing INFO or WARNING. Maybe you want to use the. You can find them here: https://github.com/goat-community/goat/blob/dev/app/api/src/utils.py

Reply:
:heavy_check_mark: Done some basics as well :smile: 
![Screenshot from 2022-08-16 00-58-14](https://user-images.githubusercontent.com/3726265/184712774-3a0c99c4-6c61-4cea-85c8-2de837d9beee.png)